common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -46,166 +40,65 @@
-    private static final String MDEVICE_NULL_ERROR_MESSAGE = "Instance UsbYubiKitDevice variable (mDevice) is null.";
-    private static final String YUBIKEY_PROVIDER = "YKPiv";
     private final Activity mActivity;
-    private final YubiKitManager mYubiKitManager;
-    private UsbYubiKeyDevice mDevice;
+    //Smartcard CBA variables
+    protected final ISmartcardCertBasedAuthManager mSmartcardCertBasedAuthManager;
     private final DialogHolder mDialogHolder;
-    //Lock to help facilitate synchronization
-    private static final Object sDeviceLock = new Object();
     //Booleans to help determine if a CBA flow is being completed so that we can emit telemetry for the results.
     private boolean mIsOnDeviceCertBasedAuthProceeding;
     private boolean mIsSmartcardCertBasedAuthProceeding;
 
     /**
-     * Holds certificate found on YubiKey and its corresponding slot.
-     */
-    public static class YubiKitCertDetails {
-        private final X509Certificate cert;
-        private final Slot slot;
-
-        /**
-         * Creates new instance of YubiKitCertDetails.
-         * @param cert Certificate found on YubiKey.
-         * @param slot PIV slot on YubiKey where certificate is located.
-         */
-        public YubiKitCertDetails(@NonNull final X509Certificate cert,
-                                  @NonNull final Slot slot) {
-            this.cert = cert;
-            this.slot = slot;
-        }
-
-        /**
-         * Gets certificate.
-         * @return certificate.
-         */
-        @NonNull
-        public X509Certificate getCertificate() {
-            return cert;
-        }
-
-
-        /**
-         * Gets PIV Slot where certificate is located.
-         * @return Slot where certificate is located.
-         */
-        @Nonnull
-        public Slot getSlot() {
-            return slot;
-        }
-    }
-
-    /**
-     * Callback which will contain code to be run upon creation of a PivSession instance.
-     */
-    public interface IPivSessionCallback {
-        /**
-         * Code depending on PivSession instance to be run.
-         * @param piv PivSession instance created from UsbSmartCardConnection.
-         */
-        void onGetSession(@NonNull final PivSession piv);
-    }
-
-    /**
-     * Creates new instance of ClientCertAuthChallengeHandler and starts usb discovery for YubiKeys.
+     * Creates new instance of ClientCertAuthChallengeHandler.
+     * A manager for smartcard CBA is retrieved, and discovery for USB devices is started.
      * @param activity current host activity.
+     * @param smartcardCertBasedAuthManager ISmartcardCertBasedAuthManager instance.
      */
-    public ClientCertAuthChallengeHandler(@NonNull final Activity activity) {
+    public ClientCertAuthChallengeHandler(@NonNull final Activity activity,
+                                          @NonNull final ISmartcardCertBasedAuthManager smartcardCertBasedAuthManager) {
```

Comment:
Want to highlight change here: directly passing `ISmartcardCertBasedAuthManager` in constructor.